core/dive.h

```diff
@@ -759,10 +759,15 @@ extern void average_max_depth(struct diveplan *dive, int *avg_depth, int *max_de
 #ifdef __cplusplus
 }
 
-/* Make pointers to dive and dive_trip "Qt metatypes" so that they can
- * be passed through QVariants. */
+/* Make pointers to dive, dive_trip and dive_table "Qt metatypes" so that they can
+ * be passed through QVariants and through QML.
+ * Note: we have to use a typedef for "struct dive_table", because MOC removes the
+ * "struct", but dive_table is already the name of a global variable, leading to
+ * compilation errors. */
+typedef struct dive_table *dive_table_ptr;
```

Comment:
and then this becomes a `dive_table_t *` and that should work, right?

Reply:
Yes, I think that should work. Will do it this way if you prefer.